### PR TITLE
chore: const spending key size

### PR DESCRIPTION
**Pull request by jowparks**
## Summary
Instead of hardcoding `32` everywhere, moved this size to a constant. It will be useful for the import/export work that @EvanJRichard and I are working on.

## Testing Plan
unittests pass
## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-rust/src/keys/mod.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ pub use view_keys::*;
 mod test;
 
 const EXPANDED_SPEND_BLAKE2_KEY: &[u8; 16] = b"Iron Fish Money ";
-
+pub const KEY_SIZE: usize = 32;
```
Comment (mat-if) [inline ironfish-rust/src/keys/mod.rs:32]: `SPEND_KEY_SIZE` or something might be a little more clear - we deal with a lot of different keys

File: ironfish-rust/src/keys/mod.rs [hunk 2]
```diff
@@ -90,8 +90,9 @@ impl SaplingKey {
         let proof_authorizing_key =
             jubjub::Fr::from_bytes_wide(&Self::convert_key(spending_key, 1));
 
-        let mut outgoing_viewing_key = [0; 32];
-        outgoing_viewing_key[0..32].clone_from_slice(&Self::convert_key(spending_key, 2)[0..32]);
+        let mut outgoing_viewing_key = [0; KEY_SIZE];
+        outgoing_viewing_key[0..KEY_SIZE]
+            .clone_from_slice(&Self::convert_key(spending_key, 2)[0..32]);
```
Comment (mat-if) [inline ironfish-rust/src/keys/mod.rs:95]:
```suggestion
            .clone_from_slice(&Self::convert_key(spending_key, 2)[0..KEY_SIZE]);
```

**Review by mat-if — APPROVED** [review]

File: ironfish-rust/src/keys/mod.rs (hunk 1 above)
Comment (jowparks) [reply ironfish-rust/src/keys/mod.rs:32]: Aren't all the keys the same size?

Comment (jowparks) [reply ironfish-rust/src/keys/mod.rs:32]: Note: yes they will most likely stay the same size

**Pull request merged**